### PR TITLE
[16.0][IMP+FIX] spreadsheet*_oca: New field to name the pivot table and fixed error adding to new sheet of dashboard

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnaiva TT50012 and TT50206

ping @pedrobaeza @chienandalu

### CODE REVIEW

**Comment by CarlosRoca13** [comment]
Added improvements to allow generate dynamic tables as showed in next gif and added an option to download the spreadsheet as xlsx (this option is available inside a spreadsheet on menu item `File`)
dynamic_table_example

**Comment by carolinafernandez-tecnativa** [comment]
Thanks for the improvements :) Some comments:

1- I have checked and when downloading XLS is not respeting spreadsheet format.

Example:

image

image

2- Dynamic field i think it must be visible only when trying to generate spreadhseet from pivot view, as if we try to create in tree view form is not working.

image

3- Database source is not filling automatically when creating spreadhseet from tree view. What should i add in this field?

image

image

**Comment by CarlosRoca13** [comment]
> 1- I have checked and when downloading XLS is not respeting spreadsheet format.

The XLSX data is generated by the core module. If you check the behavior in the odoo enterprise module, you will see that the result is the same.

For other comments I will do the changes, thanks :)

**Comment by CarlosRoca13** [comment]
@carolinafernandez-tecnativa changes done :smile: 

The value of Datasource Name is the name that will be filled to identify the table on the spreadsheet.

File: spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py [hunk 1]
```diff
@@ -42,3 +43,24 @@ def _compute_can_edit(self):
...
+            self.env["ir.model.data"]
+            .search(
+                [
+                    ("model", "=", self._name),
+                    ("module", "!=", "__export__"),
+                    ("noupdate", "=", 0),
+                ]
+            )
+            .mapped("res_id")
+        )
```
Comment (chienandalu) [inline spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: I created a brand new dashboard but I can't make it to show up in the wizard. Am I doing something wrong

Comment (CarlosRoca13) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: You have to add a spreadsheet to the panel to be able to show it

Comment (chienandalu) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: I mean, I can't select any dashboard in the wizard

Comment (CarlosRoca13) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: example

I am able to select it.

Comment (chienandalu) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: I was trying to dashboard new sheet. I was expecting to let me choose a dashboard to wich apend a sheet. Did I understand it wrong?

Comment (CarlosRoca13) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: Yes :sweat_smile: that option is to add a sheet to a created spreadsheet for the dashboard

Example:
new sheet example

This option can be used to set data that has to be hidden but it has to be used to make some operations

Comment (CarlosRoca13) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: I think that option that you say can be set, I will see how to add it 

Thanks :smile:

Comment (chienandalu) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: Ah, ok. See? I was getting it wrong :sweat_smile:

Comment (CarlosRoca13) [reply spreadsheet_dashboard_oca/models/spreadsheet_dashboard.py:63]: Added new option to create new dashboards :smile:

**Comment by CarlosRoca13** [comment]
The tables imported from the tree are dynamic by nature. So I have made some changes so that the dynamic selector is selected by default and that the value of the number of rows is filled with the value that was used in the threshold

Example:
tables

**Review by pedrobaeza — APPROVED** [review]
Tested and working.

Thanks for all the fixes and the new excellent features!

**Review by AntoniRomera — APPROVED** [review]
LGTM
We have in mind to start the migration to 17.0, if you merge this PR we could use it as a starting point.

**Review by carolinafernandez-tecnativa — APPROVED** [review]
Functional review LGTM :) Thanks!

**Pull request merged**